cardinal/config.go

```diff
@@ -160,3 +189,52 @@ func (w *WorldConfig) setLogger() error {
 
 	return nil
 }
+
+func setupViper() {
+	once.Do(func() {
+		// Bind the command-line flags to Viper
+		pflag.String(tomlConfigFileEnvVariable, "", "path to the toml config file")
+		pflag.Parse()
+		err := viper.BindPFlags(pflag.CommandLine)
+		if err != nil {
+			log.Debug().Err(err).Msg("failed to bind command-line flags to Viper")
+			// Continue even if the binding fails
+		}
+	})
+
+	// Bind the environment variables to Viper
+	viper.AutomaticEnv()
+
+	// Set default toml config file name and type
+	viper.SetConfigName("world") // name of config file (without extension)
+	viper.SetConfigType("toml")  // REQUIRED if the config file does not have the extension in the name
+
+	// Find the toml file from the flag and env variable
+	tomlFilePath := viper.GetString(tomlConfigFileEnvVariable)
+	if tomlFilePath == "" {
+		// Find the toml file in the current and parent directories
+		viper.AddConfigPath(".") // look for config in the working directory
+
+		// Add parent directory to the search path
+		parentDir, err := os.Getwd()
```

Comment:
you need to run a check first whether you've already found a world.toml file in the current directory. if there is already a config file in the current directory, you don't need to check the parent directory.

Reply:
updated